### PR TITLE
Bugfix main article can be taken down

### PR DESCRIPTION
**Pull request by maciejSamerdak**
Fixes a bug when the main article can be taken down, causing graphical disruptions at the blog page. Works as follows:
1. If an article has the main article attribute disabled, it's enabled again after save is performed.
2. If the main article is deleted, an article with the most recent date is picked as the new main article.

### CODE REVIEW

File: blog/models.py [hunk 1]
```diff
@@ -179,3 +173,25 @@ def _validate_recommended_articles_uniqueness(self) -> None:
                 raise ValidationError(message=f"'{stream_child.value}' is listed more than once!")
             else:
                 article_pages_set.add(stream_child.value)
+
+
+@receiver(post_save, sender=BlogArticlePage)
+def ensure_main_article_exists(sender: Any, instance: Any, **_kwargs: Dict[Any, Any]) -> None:
+    if not instance.is_main_article and not sender.objects.exclude(id=instance.id).filter(is_main_article=True):
```
Comment (kbeker) [inline blog/models.py:180]: Please call `exists()` method on query -> 
`not sender.objects.exclude(id=instance.id).filter(is_main_article=True).exists()`
It will give you True/False value

File: blog/models.py [hunk 2]
```diff
@@ -179,3 +173,25 @@ def _validate_recommended_articles_uniqueness(self) -> None:
                 raise ValidationError(message=f"'{stream_child.value}' is listed more than once!")
             else:
                 article_pages_set.add(stream_child.value)
+
+
+@receiver(post_save, sender=BlogArticlePage)
+def ensure_main_article_exists(sender: Any, instance: Any, **_kwargs: Dict[Any, Any]) -> None:
```
Comment (kbeker) [inline blog/models.py:179]: Is really `sender` and `instance` need to be "Any"? Won't it be always here instance of `BlogArticlePage`?

File: blog/tests/test_blog_article_page.py [hunk 3]
```diff
@@ -107,3 +108,62 @@ def test_that_article_with_no_recommended_articles_should_be_valid(self):
...
+            read_time=5,
+        )
+        new_title = "Not So Simple Article Title"
+        expected_slug = "simple-article-title"
+
+        blog_article_page.title = new_title
+        blog_article_page.save()
+
+        self.assertEqual(blog_article_page.title, new_title)
+        self.assertEqual(blog_article_page.slug, expected_slug)
```
Comment (kbeker) [inline blog/tests/test_blog_article_page.py:126]: Why is this added here in this PR?

Comment (kbeker) [reply blog/tests/test_blog_article_page.py:126]: I mean additional test is not an issue, but I'm curious why is it here?

Comment (kbeker) [reply blog/tests/test_blog_article_page.py:126]: Oh, I see you moved it. Never mind 👍

**Review by kbeker — APPROVED** [review]

File: blog/models.py (hunk 1 above)
Comment (maciejSamerdak) [reply blog/models.py:180]: Oh, I've forgotten about it. I believe I saw this being omitted somewhere in some template as well, so I probably based this code on that. Maybe I'll use this opportunity to fix those as well.

File: blog/models.py (hunk 2 above)
Comment (maciejSamerdak) [reply blog/models.py:179]: I suppose so, I guess my outlook here was too abstract. 😃 While we're on the topic here: should kwargs be described with `Dict[Any, Any]` or a more precise `Dict[str, Any]`.

File: blog/tests/test_blog_article_page.py (hunk 3 above)
Comment (maciejSamerdak) [reply blog/tests/test_blog_article_page.py:126]: Yup, while I was figuring out where to put new tests (integration or unit) I took a look at this test again, which was located in the integration tests file. I figured it's not a correct place for it after all, since the test clearly concerns the object alone. So I figured it I can move it while working on this file.

File: blog/models.py (hunk 1 above)
Comment (maciejSamerdak) [reply blog/models.py]: Done.
